### PR TITLE
Fix and extend github actions workflows

### PR DESCRIPTION
**Pull request by ruestefa**
The actions workflow `precommit` so far tested `pre-commit run -a` on the blueprint project itself, but used the requirements files of the template to create the dev env, which broke after I added `python>={{cookiecutter.python_version}}` to `{{cookiecutter.project_slug}}/requirements/requirements.txt` because the jinja template was of course still there.

Tasks:
- [x] Add requirements and environment files to the blueprint itself (with `cookiecutter` as the sole run req and `pre-commit` the sole dev req)
- [x] Adapt `.github/workflows/precommit.yml` to use those requirements files, and clean it up
- [x] Add new workflow that creates a project from the template and runs pre-commit and pytest in that

### CODE REVIEW

**Comment by ruestefa** [comment]
Adapted existing workflows a bit more (names etc.)

New workflow now does the following:
- Create new project (incl. git init etc.)
- Create dev env and install project in there
- Run pre-commit hooks (incl. black, isort, pydocstyle, flake8)
- Run the major hooks again directly to ensure they validate also on their own
- Run mypy (not in pre-commit, but should be)
- Run pylint (not in pre-commit; should it be?)
- Run pytest

File: .github/workflows/new-project.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    - name: Run linters
+      working-directory: ./flying_circus
+      run: |
+        conda run --name flying-circus-dev pydocstyle
+        conda run --name flying-circus-dev flake8
+        conda run --name flying-circus-dev pylint src tests
+    - name: Run unit tests
+      working-directory: ./flying_circus
+      run: |
+        conda run --name flying-circus-dev pytest
```
Comment (regDaniel) [inline .github/workflows/new-project.yml:75]: If I understand correctly, this triggers the creation and testing of a new "dummy" project through pre-commit. We already do this in Jenkins upon merges to the main branch. I think we should decide if we do it with Jenkins or pre-commit, but doing it twice seems to be a bit much to me.

File: .github/workflows/blueprint-pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Run pre-commit in blueprint
```
Comment (regDaniel) [inline .github/workflows/blueprint-pre-commit.yml:1]: this is mainly a renaming to make things clearer right? Seems reasonable to me.

**Review by regDaniel — CHANGES_REQUESTED** [review]
We should briefly discuss about having a package build in pre-commit and wheter this is worth having another pre-commit file in the repo as we already do this through Jenkins. But maybe I am missing something here.

File: .github/workflows/new-project.yml (hunk 1 above)
Comment (regDaniel) [reply .github/workflows/new-project.yml:75]: ah yes sorry. Anyways Jenkins is a mandatory reviewer for pushes to the main. But if it works reasonably well, we could even think about dropping the corresponding action in Jenkins. However I didn't really think this through yet and I'd also be curious on what @cosunae thinks.

**Comment by ruestefa** [comment]
Closing as changes and adaptations are already in #47 .

**Pull request closed without merging**